hugr-core/src/envelope.rs

```diff
@@ -300,6 +300,15 @@ pub enum EnvelopeError {
         #[from]
         source: crate::extension::ExtensionRegistryLoadError,
     },
+    /// The specified payload format is not supported.
+    #[error(
+        "The envelope configuration has unknown {}. Please update your HUGR version.",
+        if flag_ids.len() == 1 {format!("flag #{}", flag_ids[0])} else {format!("flags {}", flag_ids.iter().join(", "))}
```

Comment:
Nit: you're using # in the singular case. You can use it too in the plural by changing the join argument to ", #" as well as adding one to the string